Automattic/wp-calypso
# Jetpack Cloud: Add isomorphic routing for `jetpack-cloud-pricing` section

The Jetpack Cloud Pricing section is available to [logged-in and logged-out](https://github.com/Automattic/wp-calypso/blob/trunk/client/sections.js#L499) users, and accepts a language URL parameter. However, as a non-isomorphic section, the server request is being handled by [handleSectionPath](https://github.com/Automattic/wp-calypso/blob/trunk/client/server/pages/index.js#L836-L857) the URL parameter is not being taken into account when rendering the page document and therefore the script tag for the translations is missing.

The result of the missing translations script is that the page is initially always rendered in the default language, causing the English strings to be flashing for a moment, before the translations are loaded: 

https://user-images.githubusercontent.com/2722412/214534080-53e1c158-ca5e-4d89-8d93-a36e2875d556.mp4

In some cases, the translations are never being rendered, e.g. https://cloud.jetpack.com/de/pricing/jetpack-boost/welcome.

This PR adds isomorphic routing for the pricing section and introduces server routing for `/pricing` and `/plans` that applies the `setLocaleMiddleware()`. As a result, when a language parameter is present in the URL, the `context.lang` will be set accordingly and the translations script file will be loaded on the page via `<script>` tag.

#### Proposed Changes

* Add isomorphic routing for `jetpack-cloud-pricing` section.

#### Testing Instructions

* Code review.
* Checkout the branch locally or use the Jetpack Cloud Live build to test the changes.
* Go to `/{lang}/pricing` and confirm there is no longer a "english strings flashing" effect.
* Go to `/{lang}/pricing/jetpack-boost/welcome` and confirm it is now being translated into the requested Mag-16 language.

#### Pre-merge Checklist





Related to p1674163520998419-slack-CSD6UPMU4


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203527206590732
